src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIMarsConnection.cs

```diff
@@ -14,50 +14,45 @@ namespace Microsoft.Data.SqlClient.SNI
     /// </summary>
     internal class SNIMarsConnection
     {
-        private const string s_className = nameof(SNIMarsConnection);
-
-        private readonly Guid _connectionId = Guid.NewGuid();
-        private readonly Dictionary<int, SNIMarsHandle> _sessions = new Dictionary<int, SNIMarsHandle>();
-        private readonly byte[] _headerBytes = new byte[SNISMUXHeader.HEADER_LENGTH];
-        private readonly SNISMUXHeader _currentHeader = new SNISMUXHeader();
+        private readonly Guid _connectionId;
+        private readonly Dictionary<int, SNIMarsHandle> _sessions;
         private SNIHandle _lowerHandle;
-        private ushort _nextSessionId = 0;
-        private int _currentHeaderByteCount = 0;
-        private int _dataBytesLeft = 0;
-        private SNIPacket _currentPacket;
+        private ushort _nextSessionId;
 
         /// <summary>
         /// Connection ID
         /// </summary>
-        public Guid ConnectionId
-        {
-            get
-            {
-                return _connectionId;
-            }
-        }
+        public Guid ConnectionId => _connectionId;
 
         public int ProtocolVersion => _lowerHandle.ProtocolVersion;
 
+        public object DemuxerSync => this;
```

Comment:
It could but the similar properties above it that already existed are not. I prefer to use public/private even on internal classes because it indicates the items that I consider to be the public contract and private implementation details cleanly, I find internal things are often private but someone needed them and didn't want to go to the effort of making them a property or something similar. A lot of the weird coupling we have in the library in places like TdsParser and the state object come from things being exposed that possibly shouldn't have been.